### PR TITLE
Add 'ManagedZone' and related support

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1138 as a base.~~

### CODE REVIEW

**Comment by dhermes** [comment]
Ping me when #1138

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1138.

File: gcloud/dns/client.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ class Client(JSONClient):
 
     :type project: string
     :param project: the project which the client acts on behalf of. Will be
-                    passed when creating a dataset / job.  If not passed,
+                    passed when creating a zone.  If not passed,
```
Comment (dhermes) [inline gcloud/dns/client.py]: Oops

File: gcloud/dns/test_zone.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+    def _setUpConstants(self):
+        import datetime
+        import os
+        from gcloud._helpers import UTC
+
+        self.WHEN_TS = 1437767599.006
+        self.WHEN = datetime.datetime.utcfromtimestamp(self.WHEN_TS).replace(
+            tzinfo=UTC)
+        self.ZONE_ID = os.getpid()
```
Comment (dhermes) [inline gcloud/dns/test_zone.py]: I'm a little scared of `os.getpid()`. Why not just make this deterministic?

File: gcloud/dns/zone.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+"""Define API ManagedZones."""
+import six
+
+from gcloud._helpers import _datetime_from_microseconds
+from gcloud.exceptions import NotFound
+
+
+class ManagedZone(object):
+    """ManagedZone are containers for DNS resource records.
```
Comment (dhermes) [inline gcloud/dns/zone.py]: `ManagedZone are`? Plural needed? or `A ManagedZone is a`?

File: gcloud/dns/zone.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    def path(self):
+        """URL path for the zone's APIs.
+
+        :rtype: string
+        :returns: the path based on project and dataste name.
+        """
+        return '/projects/%s/managedZones/%s' % (self.project, self.name)
+
+    @property
+    def created(self):
```
Comment (dhermes) [inline gcloud/dns/zone.py]: Any reason for not calling it `creation_time`?

Comment (tseaver) [inline gcloud/dns/zone.py]: Consistency with other APIs:

```bash
$ git grep "def creation"
$ git grep "def created"
gcloud/bigquery/dataset.py:    def created(self):
gcloud/bigquery/job.py:    def created(self):
gcloud/bigquery/table.py:    def created(self):
gcloud/dns/zone.py:    def created(self):
```

File: gcloud/dns/test_zone.py (hunk 2 above)
Comment (tseaver) [inline gcloud/dns/test_zone.py]: 83fc1e9

File: gcloud/dns/zone.py (hunk 3 above)
Comment (tseaver) [inline gcloud/dns/zone.py]: 3f8f4c0

File: gcloud/dns/zone.py (hunk 4 above)
Comment (dhermes) [inline gcloud/dns/zone.py]: :+1:

**Comment by dhermes** [comment]
LGTM

**Pull request merged**